lib/builtin.t

```diff
@@ -94,6 +94,58 @@ package FetchStoreCounter {
     is(blessed(bless [], "0") ? "YES" : "NO", "NO", 'blessed in boolean context handles "0" cornercase');
 }
 
+# created_as_...
+{
+    use builtin qw( created_as_string created_as_number );
+
+    # some literal constants
+    ok(!created_as_string(undef), 'undef created as !string');
+    ok(!created_as_number(undef), 'undef created as !number');
+
+    ok( created_as_string("abc"), 'abc created as string');
+    ok(!created_as_number("abc"), 'abc created as number');
+
+    ok(!created_as_string(123),   '123 created as !string');
+    ok( created_as_number(123),   '123 created as !number');
+
+    ok(!created_as_string(1.23),   '1.23 created as !string');
+    ok( created_as_number(1.23),   '1.23 created as !number');
+
+    ok(!created_as_string([]),    '[] created as !string');
+    ok(!created_as_number([]),    '[] created as !number');
+
+    ok(!created_as_string(builtin::true), 'true created as !string');
+    ok(!created_as_number(builtin::true), 'true created as !number');
+
+    ok(builtin::is_bool(created_as_string(0)), 'created_as_string returns bool');
+    ok(builtin::is_bool(created_as_number(0)), 'created_as_number returns bool');
+
+    # variables
+    my $just_str = "def";
+    ok( created_as_string($just_str), 'def created as string');
+    ok(!created_as_number($just_str), 'def created as number');
+
+    my $just_num = 456;
+    ok(!created_as_string($just_num), '456 created as string');
+    ok( created_as_number($just_num), '456 created as number');
+
+    # variables reused
+    my $originally_str = "1";
+    my $str_as_num = $originally_str + 0;
```

Comment:
We should test that this one is actually a num too. And vice versa for `$num_str` below.

Reply:
Added